### DISCUSSION TITLE
Adding users to SharePoint groups

### DISCUSSION THREAD

**Discussion by SergeB-G** [pnp/powershell#233] [Q&A]
Greetings,

When I try to add a user to a SharePoint group using the title, I get the following:

`PS C:\Users\REDACTED\OneDrive - REDACTED\Documents\scripts\PowerShell> Get-PnPGroup`
` `
`Id Title                   LoginName  `            
`-- -----                   ---------               `
`5  REDACTED - Membres        REDACTED - Membres        `
`4  REDACTED - Visiteurs      REDACTED - Visiteurs     `
`3  Propriétaires de REDACTED Propriétaires de REDACTED`
` `
` `
` `
`PS C:\Users\REDACTED\OneDrive - REDACTED\Documents\scripts\PowerShell> Add-PnPGroupMember -LoginName "REDACTED@REDACTED.com" -Group "Propriétaires de REDACTED"`
`Add-PnPGroupMember : La référence d'objet n'est pas définie à une instance d'un objet.`
`Au caractère Ligne:1 : 1`
`+ Add-PnPGroupMember -LoginName "REDACTED@REDACTED.com" -Group "Propr ...`
`+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~`
`    + CategoryInfo          : WriteError: (:) [Add-PnPGroupMember], NullReferenceException`
`    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Principals.AddUserToGroup`

Although, the ID works fine, am I doing something incorrectly?

Please note that the error message in English would be: "Object reference not set to an instance of an object."

May someone help me out with this issue please?

Thank you for your time and help, it is greatly appreciated.

**Comment by erwinvanhunen** [comment]
First a tip

If you paste in results, simple start the with a line with three back ticks (```), then paste your results and end with 3 back ticks on a new line. 

```
This is the result 
C:> no wrapping in backticks needed
```

If you write the first line as ```powershell then it will actually use color coding:

```powershell
Connect-PnPOnline -Url https://contoso.sharepoint.com
Get-PnPList
```

Now to your issue, the only reason I can come with up that causes this error is that the user actually doesn't exist within your tenant? If you want to invite an external user you will have to provide at least the email address (and not the LoginName).

**Reply by gautamdsheth** [reply]
I have seen this happen in non-English sites, often after creating the site collection. This is due to the async nature of site creation where the group names are not yet correctly set.

What you can do is something like this based on group id and role:

```
$groups = Get-PnPGroup
$adminGroup = $groups | Where-Object {Get-PnPGroupPermissions -Identity $_.Title | 
Where-Object {$_.RoleTypeKind -eq "Administrator"}}
Add-PnPGroupMember -LoginName "REDACTED@REDACTED.com" -Group $adminGroup.Id
```
Adding the group member based on Id works across multi-lingual sites.

**Reply by SergeB-G** [reply]
Greetings,

I ended up doing something similar for testing purposes and figured out it was a language issue as well:

```
$groupName = "Propriétaires de " + $teamSite
$groupID = get-pnpgroup -Identity $groupName
$result = Add-PnPGroupMember -LoginName REDACTED@REDACTED.com -Group $groupID.Id
```

@erwinvanhunen Thank you for the reply and the information about quoting code correctly, the group and or user were created at the time of excuting the code, I ran some other tests and validated it with another site.

@gautamdsheth Unfortunately, since you replied to a post, I cannot mark it as the answer, am I missing something please?

Thank you again for your time and help.
